src/D2L.Bmx/UnixParentProcess.cs

```diff
@@ -0,0 +1,47 @@
+using System.Diagnostics;
+
+namespace D2L.Bmx;
+
+internal class UnixParentProcess {
+
+	public static string GetParentProcessName() {
+		int parentPid = GetParentProcessId( GetParentProcessId( Process.GetCurrentProcess().Id ) );
+		var proccessStartInfo = new ProcessStartInfo {
+			FileName = "/bin/ps",
+			ArgumentList = { "-p", $"{parentPid}", "-o", "comm=" },
+			RedirectStandardOutput = true,
+			UseShellExecute = false,
+			CreateNoWindow = true
+		};
+
+		using( var proc = Process.Start( proccessStartInfo ) ) {
+			if( proc is not null ) {
+				string output = proc.StandardOutput.ReadToEnd().Replace( "-", "" );
+				proc.WaitForExit();
+				return output;
+			}
+			return "bash";
+		}
+	}
+
+	private static int GetParentProcessId( int pid ) {
+		var proccessStartInfo = new ProcessStartInfo {
+			FileName = "/bin/ps",
+			ArgumentList = { "-p", $"{pid}", "-o", "ppid=" },
+			RedirectStandardOutput = true,
+			UseShellExecute = false,
+			CreateNoWindow = true,
+		};
+
+		using( var proc = Process.Start( proccessStartInfo ) ) {
+			if( proc is not null ) {
```

Comment:
nit:
* use [using declaration](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using) to reduce nesting
* use early exit to reduce nesting (i.e. if null then return first)